src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -722,6 +738,41 @@ private IConversionExpression CreateBoundAsOperatorOperation(BoundAsOperator bou
             return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IDelegateCreationExpression CreateBoundDelegateCreationExpressionOperation(BoundDelegateCreationExpression boundDelegateCreationExpression)
+        {
+            Lazy<IOperation> target = new Lazy<IOperation>(() =>
+            {
+                if (boundDelegateCreationExpression.Argument is BoundMethodGroup boundMethodGroup &&
+                    boundMethodGroup.Methods.Length == 1)
+                {
+                    // If this is a method binding, and only 1 candidate method was found, then we want to expose
+                    // this child as an IMethodBindingReference. Otherwise, we want to just delegate to the standard
+                    // CSharpOperationFactory behavior
+                    return CreateBoundMethodGroupSingleMethodOperation(boundMethodGroup, boundDelegateCreationExpression.MethodOpt, boundMethodGroup.SuppressVirtualCalls);
+                }
+                else
+                {
+                    return Create(boundDelegateCreationExpression.Argument);
+                }
+            });
+            SyntaxNode syntax = boundDelegateCreationExpression.Syntax;
+            ITypeSymbol type = boundDelegateCreationExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundDelegateCreationExpression.ConstantValue);
+            bool isImplicit = boundDelegateCreationExpression.WasCompilerGenerated;
+            return new LazyDelegateCreationExpression(target, _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private IMethodBindingExpression CreateBoundMethodGroupSingleMethodOperation(BoundMethodGroup boundMethodGroup, IMethodSymbol methodSymbol, bool suppressVirtualCalls)
+        {
+            bool isVirtual = methodSymbol != null && (methodSymbol.IsAbstract || methodSymbol.IsOverride || methodSymbol.IsVirtual) && !suppressVirtualCalls;
+            Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundMethodGroup.InstanceOpt));
+            SyntaxNode bindingSyntax = boundMethodGroup.Syntax;
+            ITypeSymbol bindingType = boundMethodGroup.Type;
+            Optional<object> bindingConstantValue = ConvertToOptional(boundMethodGroup.ConstantValue);
+            bool isImplicit = boundMethodGroup.WasCompilerGenerated;
+            return new LazyMethodBindingExpression(methodSymbol, isVirtual, instance, methodSymbol, _semanticModel, bindingSyntax, bindingType, bindingConstantValue, boundMethodGroup.WasCompilerGenerated);
```

Comment:
>methodSymbol [](start = 51, length = 12)

Why are we passing methodSymbol twice? Could it be null, what happens then?

Reply:
We're passing it twice because IMethodBindingExpression has a MethodSymbol property, with Type IMethodSymbol, and it inherits from IMemberReferenceExpression, which has a Member property with Type ISymbol. I'll deduplicate it.